broadinstitute/single_cell_portal_core
# Refactor CSFV to use chunked reads

This is mostly a pure refactor, with no behavioral changes expected, except a few noted below.  The goal was to lay the groundwork for parsing and validation of larger files with larger lines, and also make it easier to add parsing functions for specific file types.  Although many file types share common validations, making a separate function per fileType hopefully will make it easier to read and comprehend the validations that are happening, at the cost of a few extra lines of code.  

CHANGED BEHAVIOR: 
  1. Only leading and trailing quotes are now trimmed from values prior to validation, to match the behavior of ingest-pipeline.  So e.g. for a column header of `dfafe"dfaf`, the quote will be preserved.  But for a column header of `"stuff"` the quotes will be stripped.
  2. For duplicate cells, the error message will only display the first 10 duplicate cells, to avoid making the message display unreadable
  3. Minor changes to logging properties of the file.  For example, we now log 'linesRead' rather than numRows, since we may not always be reading the whole file.  We also now log 'isGzipped'

TO TEST:
  1.  run `yarn test` - confirm all tests pass
  2. in the upload wizard, experiment with adding various right/wrong files (including some from the test/test_data/validation directory).   Compare the validation messages with what is displayed on staging when you select the same files
